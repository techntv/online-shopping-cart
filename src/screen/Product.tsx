/** @jsx jsx */
import React from 'react'
import { css, jsx } from '@emotion/core'
import { IProductDetail } from "../models/product"
const titleProduct = css({
  margin: '0',
  textDecoration: 'none',
  lineHeight: '24px',
  fontSize: '12px',
  fontWeight: 700,
  padding: '4px 0',
  color: 'black',
});
const titlePrice = css({
  fontSize: '12px',
  fontWeight: 'bold',
  color: '#78ae1a',
  marginRight: '10px'
})

const titlePriceDiscount = css({
  color: '#ccc',
  textDecoration: 'line-through'
})
const imageProductHover = css({
  transition: 'transform .8s ease'
})


const imageProduct = css`
  display: flex;
  justify-content: center;
  align-items: center;
  background: rgba(204, 204, 204, 0.3);
  overflow: hidden;
  cursor: pointer;
  &:hover img{
    transform: scale(1.5)
  }
`

interface IProductProps {
  product: IProductDetail,
  handleAddToCart: Function
}
const Product = ({ product, handleAddToCart }: IProductProps) => {
  return <div className="col-md-6 col-lg-3">
    <div className="product" >
      <div className="product-image" css={[imageProduct]}>
        <img
          className="img-fluid"
          src={product.image}
          alt={product.name}
          css={imageProductHover}
        />
      </div>
      <div className="">
        <h3>
          <a href="#" css={titleProduct}>{product.name}</a>
        </h3>
        <div className="pricing d-flex justify-content-between align-items-center mb-4">
          <p className="price mb-0">
            <span className="price-sale" css={[titlePrice, titlePriceDiscount]} >{product.salePrice}</span>
            <span className={' mr-2 price-original'} css={titlePrice}>{product.originalPrice}</span>
          </p>
          <button type="button" className='btn btn-primary btn-sm' onClick={() => handleAddToCart(product)}>Add Cart</button>
        </div>

      </div>
    </div>
  </div>

}

export default Product;