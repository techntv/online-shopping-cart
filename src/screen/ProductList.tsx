/** @jsx jsx */
import * as React from 'react';
// import data from "../apiData.json"
import Product from "./Product"
import { css, jsx } from '@emotion/core'
import { connect } from 'react-redux';
import { addToCart } from '../actions/actionsCart';
import { fetchProducts } from '../actions/actionsProduct'
import { IProductDetail } from "../models/product"
const productListStyle = css({
  marginTop: '60px'
})

class ProductList extends React.Component<any, any> {
  state = {
    productList: this.props.items
  }

  handleAddToCart = (product: any) => {
    this.props.addToCart(product)
  }

  componentDidMount() {
    this.props.fetchProducts()
  }
  render() {
    let { productList } = this.state;
    let { from, to, total, items } = this.props;
    return <div className="container-fluid">
      <div className="row">
        <div className="col col-xs-12 col-md-3">

        </div>
        <div className="col col-xs-12 col-md-9">
          <div className="my-4">
            <p>Hiển thị {from}-{to} trên {total} </p>
          </div>
          <div className="product-list row" css={productListStyle}>
            {
              items.map((product: IProductDetail, index: number) => {
                return <Product product={product} key={product.name + index} handleAddToCart={this.handleAddToCart} />
              })
            }
          </div>
        </div>
      </div>
    </div>
  }
}

const mapDispatchToProps = {
  addToCart,
  fetchProducts
}

const mapStateToProps = (state: any, ownProps: any) => {
  const { products: { total, pageSize, page, items } } = state
  return {
    total,
    pageSize,
    page,
    from: (page * pageSize) - pageSize,
    to: page * pageSize,
    items
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ProductList);