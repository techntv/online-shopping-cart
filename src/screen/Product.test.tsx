import renderer from 'react-test-renderer'
import ProductList from './ProductList'
import Product from './Product';
import * as React from 'react'
const mockProduct = {
  "id": "0001",
  "name": "Winter Coat from France",
  "image": "https://api.demo.nordiccoder.com/images/product_1.png",
  "thumbnail": "https://api.demo.nordiccoder.com/images/product_1.png",
  "shortDescription": "product 1",
  "category": "men",
  "salePrice": 520,
  "originalPrice": 590,
  "images": [
    "https://api.demo.nordiccoder.com/images/product_1_1.png",
    "https://api.demo.nordiccoder.com/images/product_1_2.png"
  ],
  "thumbnails": [
    "https://api.demo.nordiccoder.com/images/product_1_1.png",
    "https://api.demo.nordiccoder.com/images/product_1_2.png"
  ]
}

test('Product component', () => {
  const component = renderer.create(
    <Product product={mockProduct} handleAddToCart={() => { }} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})