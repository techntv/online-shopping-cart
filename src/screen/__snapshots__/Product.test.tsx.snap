// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component 1`] = `
<div
  className="col-md-6 col-lg-3"
>
  <div
    className="product"
  >
    <div
      className="product-image css-f9ikke-Product"
    >
      <img
        alt="Winter Coat from France"
        className="img-fluid css-y78e7o-Product"
        src="http://api.demo.nordiccoder.com/images/product_1.png"
      />
    </div>
    <div
      className=""
    >
      <h3>
        <a
          className="css-acl8hv-Product"
          href="#"
        >
          Winter Coat from France
        </a>
      </h3>
      <div
        className="pricing d-flex justify-content-between align-items-center mb-4"
      >
        <p
          className="price mb-0"
        >
          <span
            className="price-sale css-1veg0n1-Product"
          >
            520
          </span>
          <span
            className=" mr-2 price-original css-6pi4qh-Product"
          >
            590
          </span>
        </p>
        <button
          className="btn btn-primary btn-sm"
          onClick={[Function]}
          type="button"
        >
          Add Cart
        </button>
      </div>
    </div>
  </div>
</div>
`;
